expts/main_run_finetuning_v2.py

```diff
@@ -0,0 +1,153 @@
+# General imports
+import argparse
+import os
+from os.path import dirname, abspath
+import yaml
+from copy import deepcopy
+from omegaconf import DictConfig
+import timeit
+from loguru import logger
+from datetime import datetime
+import torch.nn as nn
+from lightning.pytorch.utilities.model_summary import ModelSummary
+
+from graphium.utils.mup import set_base_shapes
+from graphium.finetuning import modify_cfg_for_finetuning, MolecularFinetuning
+
+# Current project imports
+import graphium
+from graphium.config._loader import (
+    load_datamodule,
+    load_metrics,
+    load_architecture,
+    load_predictor,
+    load_trainer,
+    save_params_to_wandb,
+    load_accelerator,
+    load_yaml_config,
+)
+from graphium.utils.safe_run import SafeRun
+from graphium.utils.command_line_utils import update_config, get_anchors_and_aliases
+
+
+# WandB
+import wandb
+
+# Set up the working directory
+MAIN_DIR = dirname(dirname(abspath(graphium.__file__)))
+
+CONFIG_FILE = "expts/finetune_configs/config_toy_finetuning_v2.yaml"
+
+os.chdir(MAIN_DIR)
+
+
+def main(cfg: dict, run_name: str = "main", add_date_time: bool = True) -> None:
+    st = timeit.default_timer()
+
+    date_time_suffix = ""
+    if add_date_time:
+        date_time_suffix = datetime.now().strftime("%d.%m.%Y_%H.%M.%S")
+
+
+
+    ########################
+    # Changes for finetuning
+
+    # Modify config
+    cfg = modify_cfg_for_finetuning(cfg)
+
+    ########################
+
+
+    wandb.init(entity="multitask-gnn", project=cfg["constants"]["name"], config=cfg)
+
+    # Initialize the accelerator
+    cfg, accelerator_type = load_accelerator(cfg)
+
+    # Load and initialize the dataset
+    datamodule = load_datamodule(cfg, accelerator_type)
+
+    # Initialize the network
+    model_class, model_kwargs = load_architecture(
+        cfg,
+        in_dims=datamodule.in_dims,
+    )
+
+    datamodule.prepare_data()
+
+    metrics = load_metrics(cfg)
+    logger.info(metrics)
+
+    predictor = load_predictor(
+        cfg, model_class, model_kwargs, metrics, accelerator_type, datamodule.task_norms
```

Comment:
I would copy the `model_kwargs` from the `pretrained_model`, but modify the configs with a similar logic to `modify_cfg_for_pretrained`